### PR TITLE
Fix Tailwind line heights

### LINKED ISSUE #477: Line heights are incorrectly configured in Tailwind
## Description

The Tailwind configuration for line-heights is ignored because the key is supposed to be [`lineHeight` (not plural)](https://tailwindcss.com/docs/line-height#customizing-your-theme). This means our custom config is ignored and the built-in line heights are used.

https://github.com/WordPress/openverse-frontend/blob/9ba8cd356689ce81e0153592b29ace57d866c236/tailwind.config.js#L170

A little bit of this fix has been started in WordPress/openverse-frontend#2060 (relevant [diff](https://github.com/WordPress/openverse-frontend/pull/2060/files#diff-ddc1067f080930bc48865f4660316a14b21bbc9f9ba85975ce8479342b983d2eR170) and [diff](https://github.com/WordPress/openverse-frontend/pull/2060/files#diff-ddc1067f080930bc48865f4660316a14b21bbc9f9ba85975ce8479342b983d2eR190-R192)) where it was first spotted.

## Additional context

It should be addressed separately as it will break almost every existing visual regression snapshot (hence the label).

### LINKED ISSUE #947: Line heights are inconsistent
## Description

Tailwind config parameter for setting custom line heights is called `lineHeight`. In our `tailwind.config` it is called `lineHeights` (with an extra s at the end). Because of this, all the named line heights (larger, large, normal, snug, tight) fall back on the Tailwind default values instead of using our custom values. This makes our line heights inconsistent.

We need to rename `lineHeights` to `lineHeight` in `tailwind.config` and update all the snapshots.

## Reproduction

Check the homepage h1 element. Instead of using our `tight` value of 1.2 for line height, it uses the Tailwind default of 1.25.

## Additional context
There is a `FIXME` comment about it in the code. We did not update it when developing the new header/footer/homepage to prevent a massive update in snapshots. Now, it should be useful to update before the `Core UI improvements` project.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #947 by @obulat 
Fixes #477 by @dhruvkb 

## Description

This PR moves the line height settings to `extend` and adjust them so that only the set values are used, and not the Tailwind default fallback. This means that a lot of snapshots need to be updated because in many places, the line height was set to 1.375 or 1.25 instead of 1.3 instead of 1.3 as in Openverse mockups.
You can see the current values for named line heights in the Tailwind viewer: https://wordpress.github.io/openverse/tailwind/#Line%20Height (Compare to this PR: https://wordpress.github.io/openverse/_preview/946/tailwind/#Line%20Height)

Screenshot 2023-03-19 at 7 30 33 AM

This PR is important before the work on Core UI improvements because the line height of the buttons is also affected. We could also add this to the scope of Core UI improvements project.

## Testing Instructions

Check the snapshots - they should look fairly similar, but the heights should change slightly. Everything should build fine (no errors of `class` not found for tailwind build step)

## Checklist

- [x] My pull request has a descriptive title (not a vague title like
      `Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or
      a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible
      errors.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by panchovm** [comment]
Yesterday I shared [a concern](https://wordpress.slack.com/archives/C02012JB00N/p1679412051217449?thread_ts=1679411849.316949&cid=C02012JB00N) that this PR might conflict with what we understand as a source of truth for designs, but I understood the PR scope incorrectly and now I see what you @obulat were talking about.

My idea for revisiting the text styles is making consistent five heigh-line cases we currently have.
1. A default value in buttons and other interactive UI components.
2. A value for headings where the text length is short but might take a significant portion of the screen.
3. A value for descriptive texts, like texts in the report content box.
4. A value for long paragraphs, like internal pages where text is the main content.
5. A value for more "decorative" sections, like H1 in the homepage and all content results.

So based on the slack thread message, we are addressing this problem first, and then defining the values for the cases mentioned above.

**Comment by obulat** [comment]
> So based on the slack thread message, we are addressing this problem first, and then defining the values for the cases mentioned above.

Yes, we will be able to fully address the cases you list _after_ fixing the code for line heights in this PR!
After this PR is merged, we will know which values are used where, instead of wondering whether the line height is set by Tailwind default or by the value we added.

**Comment by zackkrida** [comment]
@obulat is this ready for review?

**Comment by obulat** [comment]
> @obulat is this ready for review?

Yes

File: frontend/src/styles/tailwind.css [hunk 1]
```diff
@@ -45,17 +44,17 @@ Heading 3           - 28px - 5xl bold
...
-Label Bold          - 13px - sm semibold
-Label Regular       - 13px - sm regular
-Caption Regular     - 12px - sr regular
-Caption Bold        - 12px - sr semibold
-Category            - 12px - sr semibold UPPERCASE
-Time                - 11px - xs semibold (default leading-[120%])
+Paragraph Large     - 16px - 2xl  regular  leading-larger
+Paragraph Small     - 14px - base regular  leading-large
+Body                - 14px - base regular  (default leading-normal)
+Description Regular - 14px - base regular  leading-snug
```
Comment (zackkrida) [inline frontend/src/styles/tailwind.css:50]:
```suggestion
Description Regular - 14px - base regular leading-snug
```

**Review by zackkrida — APPROVED** [review]
It's disappointing that the snapshots change but it indeed only looks like small line height adjustments.

**Review by dhruvkb — APPROVED** [review]
LGTM. All of the snapshots moved by a few pixels but everything still looks just right.

**Pull request merged**